dolphin/alert_manager/trap_receiver.py

```diff
@@ -136,32 +233,21 @@ def _cb_fun(self, state_reference, context_engine_id, context_name,
             # Unexpected exception occurred
             LOG.error(e)
 
-    def _snmp_v2v3_config(self):
-        """Configures snmp v2 and v3 user parameters."""
-        community_str = constants.SNMP_COMMUNITY_STR
-        config.addV1System(self.snmp_engine, community_str, community_str)
-        auth_priv_protocols = {
-            'usmHMACMD5AuthProtocol': config.usmHMACMD5AuthProtocol,
-            'usmHMACSHAAuthProtocol': config.usmHMACSHAAuthProtocol,
-            'usmAesCfb128Protocol': config.usmAesCfb128Protocol,
-            'usmAesCfb256Protocol': config.usmAesCfb256Protocol,
-            'usmAesCfb192Protocol': config.usmAesCfb192Protocol,
-            'usmDESPrivProtocol': config.usmDESPrivProtocol,
-            'usmNoAuthProtocol': config.usmNoAuthProtocol,
-            'usmNoPrivProtocol': config.usmNoPrivProtocol
-        }
-        config.addV3User(
-            self.snmp_engine, userName=constants.SNMP_USM_USER,
-            authKey=constants.SNMP_V3_AUTHKEY,
-            privKey=constants.SNMP_V3_PRIVKEY,
-            authProtocol=auth_priv_protocols.get(
-                constants.SNMP_V3_AUTH_PROTOCOL, config.usmNoAuthProtocol),
-            privProtocol=auth_priv_protocols.get(
-                constants.SNMP_V3_PRIV_PROTOCOL, config.usmNoPrivProtocol),
-            securityEngineId=v2c.OctetString(
-                hexValue=constants.SNMP_ENGINE_ID))
-
-        return
+    def _load_snmp_config(self):
+        """Load snmp config from database when service start."""
+        ctxt = context.get_admin_context()
+        marker = None
+        finish = False
+        while not finish:
+            alert_sources = db_api.alert_source_get_all(ctxt, marker=marker,
+                                                        limit=1000)
+            for alert_source in alert_sources:
+                snmp_config = dict()
+                snmp_config.update(alert_source)
+                self._add_snmp_config(ctxt, snmp_config)
+                marker = alert_source['storage_id']
+            if len(alert_sources) < 1000:
```

Comment:
Keep 1000 as constant
Also, Please provide reason as why 1000? provide reason

Reply:
I see it being incorporated. Thanks

Reply:
Used constant instead.  Normally, alert_config should not be too much, it will rarely exceeds 1000, use this limitation for very very corner case, to check if need query again.